base/process.jl

```diff
@@ -72,6 +72,20 @@ backticks, e.g.
 """
 Cmd
 
+function hash(x::Cmd)
+    result = hash(x.exec)
+    result = result * 31 + hash(x.env)
+    result = result * 31 + hash(x.flags)
+    result = result * 31 + hash(x.dir)
+    result * 31 + hash(x.ignorestatus)
+end
```

Comment:
idiom that I've used heavily when writing Java hashCode overrides (http://stackoverflow.com/questions/299304/why-does-javas-hashcode-in-string-use-31-as-a-multiplier/299748#299748)...wasn't aware of the ability to chain two-argument hashing until now. Will switch over to using that, thanks.